.gitlab/deps_fetch.yml

```diff
@@ -1,14 +1,16 @@
 ---
 # deps_fetch stage
-# Contains jobs which fetch go dependencies for various Linux platforms, in order
+# Contains jobs which fetch go dependencies, in order
 # to reuse them in further jobs that need them.
 
 .retrieve_linux_go_deps:
```

Comment:
Since we reuse Windows runners across multiple job runs, we have to clean up everything we do on the runner itself, therefore doing as many operations in the container as possible is better. Otherwise we'd have to add another cleanup step to all Windows jobs that use the go deps, to remove the modcache directory.